src/include/mutex.h

```diff
@@ -1,11 +1,13 @@
 #ifndef CMARK_MUTEX_H
 #define CMARK_MUTEX_H
 
+#include "cmark-gfm_config.h"
+
 #include <stdbool.h>
 
-#ifdef CMARK_THREADING
+#if CMARK_THREADING
 
-#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
+#if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__)) || defined(__wasi__)
 #include <unistd.h>
 #endif
```

Comment:
At this point i'm starting to wonder if we can just get away with just checking for the existence of `unistd.h` instead of probing for different compilation environments:

```suggestion
#if defined(__has_include)
#if __has_include(<unistd.h>)
#include <unistd.h>
#endif
#endif
```

It looks like both Clang and GCC have the `__has_include` macro, so this should probably cover our bases for any other new platform over time.